src/main/java/org/fundacionjala/coding/maria/Kata3.java

```diff
@@ -0,0 +1,26 @@
+package org.fundacionjala.coding.maria;
+
+/**
+ * test.
+ */
+public class Kata3 {
+
+    private static final int LIM = 10;
+    /**
+     * @param num numero
+     * @return total persistent
+     */
+    public int persistence(int num) {
+
+        int cont = 0;
+        while (num / LIM > 0) {
+            int[] digits = Integer.toString(num).chars().map(c -> c -= '0').toArray();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment out of this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)